synapse/storage/monthly_active_users.py

```diff
@@ -186,9 +207,7 @@ def upsert_monthly_active_user(self, user_id):
                 "timestamp": int(self._clock.time_msec()),
             },
         )
-        if is_insert:
-            self.user_last_seen_monthly_active.invalidate((user_id,))
```

Comment:
I'm a bit worried about the footgun of `upsert_monthly_active_user` and `upsert_monthly_active_user_txn` being different. Is there a good reason to avoid doing this during init?

Reply:
After IRL agreed that this was the best approach - have recorded in a comment